test/classes-additional.coffee

```diff
@@ -0,0 +1,301 @@
+# Classes
+# -------
+
+# * Class Instantiation
+# * Inheritance and Super
+# * Default parameters
+# * Version Compatability
+test "multiple super calls", ->
+  # class A 
+  #   constructor: (@drink) ->
+  #   make: -> "Making a #{@drink}"
+
+  # class MultiSuper extends A
+  #   constructor: (drink) ->
+  #     super(drink)
+  #     super(drink)
+  #     @newDrink = drink
+  # eq (new MultiSuper('Late')).make(), 'Making a Late'
+
+test "@ params", ->
+  class A 
+    constructor: (@drink, @shots, @flavor) ->
+    make: -> "Making a #{@flavor} #{@drink} with #{@shots} shot(s)"
+
+  a = new A('Machiato', 2, 'chocolate')
+  eq a.make(),  "Making a chocolate Machiato with 2 shot(s)"
+
+  class B extends A
+  b = new B('Machiato', 2, 'chocolate')
+  eq b.make(),  "Making a chocolate Machiato with 2 shot(s)"
+
+test "@ params with defaults", ->
+  class A 
+    # Multiple @ params with defaults
+    constructor: (@drink = 'Americano', @shots = '1', @flavor = 'caramel') ->
+    make: -> "Making a #{@flavor} #{@drink} with #{@shots} shot(s)"
+
+  a = new A()
+  eq a.make(),  "Making a caramel Americano with 1 shot(s)"
+
+test "@ params with class params", ->
+  class Beverage
+    drink: 'Americano'
+    shots: '1'
+    flavor: 'caramel'
+
+  class A 
+    # Class creation as a default param with `this`
+    constructor: (@drink = new Beverage()) ->
+  a = new A()
+  eq a.drink.drink, 'Americano'
+
+  beverage = new Beverage
+  class B 
+    # class costruction with a default external param
+    constructor: (@drink = beverage) ->
+
+  b = new B()
+  eq b.drink.drink, 'Americano'
+
+  class C 
+    # Default constructor with anonymous empty class
+    constructor: (@meta = class) ->
+  c = new C()
+  ok c.meta instanceof Function
+
+test "@ params without super, including errors", ->
+  class A 
+    constructor: (@drink) ->
+    make: -> "Making a #{@drink}"
+  a = new A('Machiato')
+  eq a.make(),  "Making a Machiato"
+
+  throwsB = """
```

Comment:
Maybe we could reuse [assertErrorFormat](https://github.com/jashkenas/coffeescript/blob/a1afdf543763f32858f0e54c77b203eb387790d8/test/error_messages.coffee#L7) for this kinda stuff, somehow.

Also, I know you're trying to make representative examples, but in this case the test introduces lots of scope for false negatives (e.g., even without the `@` parameter this will still throw).